src/util/support/utf8_conv.c

```diff
@@ -430,7 +433,7 @@ krb5int_ucs2lecs_to_utf8s(const unsigned char *ucs2les,
     if (ucs2leslen > SSIZE_MAX)
         return ERANGE;
 
-    len = k5_ucs2s_to_utf8s(NULL, (krb5_ucs2 *)ucs2les, 0,
+    len = k5_ucs2s_to_utf8s(NULL, ucs2_of_utf8(ucs2les), 0,
```

Comment:
Test suite calls into `k5_utf8s_to_ucs2s()` eight times, and `k5_ucs2s_to_utf8s()` once.  I have also not tested this code on a big endian machine.  I did have a version with k5buf, but it's longer and more involved; the ucs2 converter function really wants to operate on the raw string, so the abstraction doesn't gain much.